tests/parsing/FilterHelperTest.py

```diff
@@ -0,0 +1,52 @@
+import unittest
+
+from coalib.parsing.FilterHelper import (
+    apply_filter,
+    apply_filters,
+    is_valid_filter,
+    _filter_section_bears
+)
+from coalib.parsing.filters import available_filters
+from coalib.parsing.InvalidFilterException import InvalidFilterException
+from coalib.settings.ConfigurationGathering import get_all_bears
+
+
+class FilterHelperTest(unittest.TestCase):
+
+    def test_apply_filter_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filter('unknown', ['args'])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_apply_filters_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filters([['language', 'C', 'Python'],
+                           ['unknown', 'args']])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_is_valid_filter(self):
+        result_for_wrong_filter = is_valid_filter('check')
+        self.assertFalse(result_for_wrong_filter)
+        result_for_right_filter = is_valid_filter('can_detect')
+        self.assertTrue(result_for_right_filter)
+
+    def test_filter_section_bears(self):
+        local_bears, global_bears = get_all_bears()
+        filter_args = {'c', 'java'}
+        result = _filter_section_bears(
```

Comment:
@sangamcse this `result` can also be somewhat meaningful like you did for the above two results which were `result1` and `result2`. We can use the name of the variable like `filter_section_result`. Any meaningful name of the variable. As you mentioned that the code is more readable if the variable names are more readable and understandable. :wink:

Reply:
Done.